nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/SystemDiagnosticsSnapshotDTO.java

```diff
@@ -361,6 +371,12 @@ public SystemDiagnosticsSnapshotDTO clone() {
             contentRepoStorageUsage.add(usage.clone());
         }
 
+        final Set<StorageUsageDTO> provenanceRepoStorageUsage = new LinkedHashSet<>();
```

Comment:
Can we add a null checks around these parts of the `clone` method?

`if (getContentRepositoryStorageUsage() != null) {`
`if (getProvenanceRepositoryStorageUsage() != null) {`
`if (getGarbageCollection() != null) {`

I realize that a couple of these weren't part of your PR but could lead to an NPE if the DTO isn't fully populated at time of invocation.